client/src/js/services/grid/TransactionService.js

```diff
@@ -295,6 +295,26 @@ function TransactionService($timeout, util, uiGridConstants, bhConstants, Notify
     return true;
   };
 
+
+  /**
+   * @method validateTransaction
+   *
+   * @description
+   * This method is used to check if a transaction is balanced by their record
+   */
+  function validateTransaction(entity) {
+    var dataEntity = entity.data.data;
+    var debit = 0,
+      credit = 0;
+    
+    dataEntity.forEach(function (journal) {
```

Comment:
A transaction is made of rows, not journals (a journal contains transactions), so a better name for this variable is `row`.  Or `line`.  Or `item`.  But `journal` is confusing.